portal-kernel/src/com/liferay/portal/kernel/exception/LocaleException.java

```diff
@@ -56,31 +59,113 @@ public LocaleException(int type, Throwable throwable) {
 		_type = type;
 	}
 
+	public Collection<String> getSourceAvailableLanguageIds() {
+		if (_sourceAvailableLanguageIds != null) {
+			return _sourceAvailableLanguageIds;
+		}
+
+		_sourceAvailableLanguageIds = _languageIdsFromLocales(
```

Comment:
@marianoalvarosaiz Here _sourceAvailableLanguageIds is used as a cache of _sourceAvailableLocales, which means you need to null it out when _sourceAvailableLocales is changed.

Same thing to _sourceAvailableLocales, it can be used as a cache too, which means you need to null it out when necessary.